src/Core/Framework/App/Context/Gateway/AppContextGatewayResponse.php

```diff
@@ -0,0 +1,48 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Framework\App\Context\Gateway;
+
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Struct\Struct;
+
+/**
+ * @phpstan-type ContextGatewayCommand array{command: string, payload: array<mixed>}
```

Comment:
why not using a DTO instead?

Reply:
@mitelg it is an array after deserialization which is only used in this internal app context. I do not see the benefit of making a DTO 🤔

Reply:
it would make sure, that all properties have the correct type after deserialization :slightly_smiling_face:

Reply:
that happens one step later, when these Commands get put into DTOs,